### ISSUE TITLE
[CR] figure out "dynamic" eye encumbrance for eyewear

### ISSUE THREAD

**Issue by qoreQyaS** [CleverRaven/Cataclysm-DDA#65131]
### Is your feature request related to a problem? Please describe.

TL/DR: what existing data (from the json files) is necessary to calculate eye encumbrance for eyewear?
this is meant as some sort of brainstorming, so please, blast out your ideas.

non-TL/DR:
i was looking for the least encumbering way to get glare protection as a vision unimpaired PC.
`pair of sunglasses` have eye encumbrance 1.
`pair of bifocal sunglasses` have no eye encumbrance and have no downside for a PC who is neither nearsighted nor farsighted.
this led to #65018: i tried to remove eye encumbrance from sunglasses which doesn't seem to reflect reality.
i found out about #57243 which introduced a weight dependent system for head encumbrance for helmets. this seems like something worthwhile implementing for exclusive eyewear.
my try to summarize how to system forks for headwear is in https://github.com/CleverRaven/Cataclysm-DDA/pull/65018#issuecomment-1508436468

#### general/decisions
1. decide which existing json data should be taken into consideration for the calculation of encumbrance for items which cover `eyes` exclusively 
    - [x] `weight`
    - [ ] `volume`
    - [x] `flags`
    - [x] `qualities`
    - [ ] `materials`
2. some items have flag `FLASH_PROTECTION` and quality `[ "GLARE" : "x" ]` what should happen here? should those be mutually exclusive for the encumbrance calculation?

#### coding
1. unhardcode get the `encumbrance_modifiers` by adding the data to a new list somewhere near https://github.com/CleverRaven/Cataclysm-DDA/blob/40b1d9db0803890a047b97db7c3cdb219e1a983b/data/json/body_parts.json#L143 and by modifiying https://github.com/CleverRaven/Cataclysm-DDA/blob/40b1d9db0803890a047b97db7c3cdb219e1a983b/src/itype.cpp#L398 to use these
3. add the ability to map `flags` to `encumbrance_modifiers`. (i.e. flag `SUN_GLASSES` adds a `FLAT:1` ), having a table in body_parts.json with the corresponging values.
4. same goes for qualities (i.e. quality `GLARE`  adds a `FLAT:10` per its value. so `[ "GLARE", 1 ]` adds 10, `[ "GLARE", 2 ]` adds 20 and so on)

#### data assembly
- get data for `encumbrance_per_weight` table.
- get data for `encumbrance_modifiers` table.
    - "special" values like `IMBALANCED` or `WELL_SUPPORTED` for helmets which have no meaning otherwise ingame so far.
    - `FLAGS` to `encumbrance_modifiers`
    - `qualities` to `encumbrance_modifiers`

### Describe alternatives you have considered.

- tidy up the eye encumbrance data for eyewear in the json files.
- add penalties for the non vision impaired PCs which use glasses/lenses for those who are. this will be done separately.
